File: hw0/Baby_Hook/sol.c
#include <stdio.h>

int __libc_start_main(
    int (*main)(int, char **, char **),
    int argc,
    char **argv,
    int (*init)(int, char **, char **),
    void (*fini)(void),
    void (*rtld_fini)(void),
    void *stack_end){
    FILE *file;
    file = fopen("./flag.txt", "r");
    if (file == NULL) {
        printf("Unable to open the file.\n");
        return 1;
    }
    char line[100];
    while (fgets(line, sizeof(line), file) != NULL) {
        printf("%s", line);
    }
    fflush(0);
    fclose(file);
    _Exit(1);
}

File: AIS3/crypto/side_channel_attack/sol.py
with open('d_rev.txt', 'r') as f:
    lines = f.readlines()
    lines = [line.strip() for line in lines]

d = 0
for line in lines:
    if "r = r * a % c" in line:
        d += 1
    elif "b >>= 1" in line:
        d <<= 1
        
print(d)